docs/Tutorial-IntegrationWithExistingApps.md

```diff
@@ -0,0 +1,745 @@
+6. Profit!
+
+<block class="android" />
+
+The keys to integrating React Native components into your iOS application are to:
+
+1. Understand what React Native components you want to integrate.
+2. Install `react-native` in your Android application root directory to create `node_modules/` directory.
+3. Create your actual React Native components in JavaScript - either all directly in a `index.android.js` file, or in a `index.android.js` file and multiple other files.
+4. Add `com.facebook.react:react-native:+` and a `maven` pointing to the `react-native` binaries in `node_nodules/` to your `build.gradle` file.
+4. Create a custom React Native specific `Activity` that creates a `ReactRootView`.
+5. Start the React Native server and run your native application.
+6. Profit!
+
+<block class="objc swift android" />
+
+## Prerequisites
+
+<block class="android" />
+
+The [Android Getting Started guide](/react-native/docs/getting-started.html) will install the appropriate prerequisites (e.g., `npm`) for React Native on the Android target platform and your chosen development environment.
+
+<block class="objc swift" />
+
+### General
+
+First, follow the [Getting Started guide](/react-native/docs/getting-started.html) for your development environment and the iOS target platform to install the prerequisites for React Native.
+
+### CocoaPods
+
+[CocoaPods](http://cocoapods.org) is a package management tool for iOS and Mac development. We use it to add the actual React Native framework code locally into your current project.
+
+```bash
+$ sudo gem install cocoapods
+```
+
+## Our Sample App
+
+<block class="objc" />
+
+Assume the [app for integration](https://github.com/JoelMarcey/iOS-2048) is a [2048](https://en.wikipedia.org/wiki/2048_(video_game) game. Here is what the main menu of the native application looks like without React Native.
+
+<block class="swift" />
+
+Assume the [app for integration](https://github.com/JoelMarcey/swift-2048) is a [2048](https://en.wikipedia.org/wiki/2048_(video_game) game. Here is what the main menu of the native application looks like without React Native.
+
+<block class="objc swift" />
+
+![Before RN Integration](img/react-native-existing-app-integration-ios-before.png)
+
+## Package Dependencies
+
+React Native integration requires both the React and React Native node modules. The React Native Framework will provide the code to allow your application integration to happen.
+
+
+### `package.json`
+
+We will add the package dependencies to a `package.json` file. Create this file in the root of your project if it does not exist.
+
+> When "root of project" is mentioned in any step, that is the location where your application's project file is located (e.g., `.xcodeproj`).
+
+Below is an example of what your `package.json` file should minimally contain.
+
+> Version numbers will vary according to your needs. Normally the latest versions for each will be sufficient.
+
+<block class="objc" />
```

Comment:
Yeah, I thought about that -- but I wanted it to match the code on GitHub that I link to at the end as much as possible.